src/validators/domain.py

```diff
@@ -1,20 +1,39 @@
 """Domain."""
 
 # standard
+from os import environ
 from pathlib import Path
 import re
+from typing import Optional, Set
 
 # local
 from .utils import validator
 
 
-def _iana_tld():
-    """Load IANA TLDs as a Generator."""
-    # source: https://data.iana.org/TLD/tlds-alpha-by-domain.txt
-    with Path(__file__).parent.joinpath("_tld.txt").open() as tld_f:
-        _ = next(tld_f)  # ignore the first line
-        for line in tld_f:
-            yield line.strip()
+class _IanaTLD:
+    """Read IANA TLDs, and optionally cache them."""
+
+    _full_cache: Optional[Set[str]] = None
```

Comment:
Just wondering, why not use `_full_cache: Set[str] = set()`?

Reply:
I think it's cleaner to have a null value representing the cache not being loaded, vs. loaded and empty. And I'm averse to creating objects that will never be used.